spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,146 @@
+from spidermon import monitors, MonitorSuite, Monitor
+from spidermon.exceptions import NotConfigured
+from ..monitors.mixins.spider import SpiderMonitorMixin
+
+SPIDERMON_MIN_ITEMS = 'SPIDERMON_MIN_ITEMS'
+SPIDERMON_MAX_ERRORS = 'SPIDERMON_MAX_ERRORS'
+SPIDERMON_EXPECTED_FINISH_REASONS = 'SPIDERMON_EXPECTED_FINISH_REASONS'
+SPIDERMON_UNWANTED_HTTP_CODES = 'SPIDERMON_UNWANTED_HTTP_CODES'
+
+
+class BaseScrapyMonitor(Monitor, SpiderMonitorMixin):
+    longMessage = False
+
+    def get_settings(self, name, default=None):
+        """Return a settings prioritized by:
+
+            #. ``{name}`` in the ``crawler.settings`` object
+            #. If ``crawler.settings.{name}`` is a dict, try to get the
+               ``{spider.name}`` key of that dictionary, this allows you to
+               have a different configuration per spider in a settings level,
+               like:
+
+               .. code-block:: python
+
+                   SPIDERMON_MAX_ERRORS = {
+                        'my_spider': 100,
+                        'other_spider': 10,
+                   }
+
+        """
+        settings_value = self.crawler.settings.get(name.upper())
+        if type(settings_value) == dict:
+            settings_value = settings_value.get(self.spider.name, None)
+        return settings_value or default
+
+
+@monitors.name('Extracted Items Monitor')
+class ItemCountMonitor(BaseScrapyMonitor):
+    """Check if spider extracted the minimum number of items.
+    You can configure it using ``SPIDERMON_MIN_ITEMS`` setting.
+    There's **NO** default value for this setting, if you try to use this
+    monitor without setting it, it'll raise a ``NotConfigured`` exception.
+    """
+    def run(self, result):
+        self.minimum_threshold = self.get_settings(SPIDERMON_MIN_ITEMS)
+        if not self.minimum_threshold:
+            raise NotConfigured('You should specify a minimum number of items '
+                                'to check against.')
+        return super(ItemCountMonitor, self).run(result)
+
+    @monitors.name('Should extract the minimum amount of items')
+    def test_minimum_number_of_items(self):
+        item_extracted = getattr(self.stats, 'item_scraped_count', 0)
+        msg = 'Extracted {} items, the expected minimum is {}'.format(
+            item_extracted, self.minimum_threshold)
+        self.assertTrue(
+            item_extracted >= self.minimum_threshold, msg=msg
+        )
+
+
+@monitors.name('Log Monitor')
```

Comment:
This is not a "Log monitor". I believe a better name would be "Number of error monitor"

Reply:
Another option is `ErrorCountMonitor`